### DISCUSSION TITLE
Extend toString to optionally render explicit multiplication as implicit multiplication

### DISCUSSION THREAD

**Discussion by btepe** [josdejong/mathjs#3067] [Ideas]
Hi,
I am trying to remove implicit multiplication signs from an expression, but I just can't get it to work like in the docs.
Here is a basic example that I tried:
`const nodeTest = math.parse('2*a');`
`console.log(nodeTest.toString({implicit: 'hide'}));`

- expected output: `2 a`
- actual output: `2 * a`

It seems to work the other way around:
`const nodeTest = math.parse('2a');`
`console.log(nodeTest.toString({implicit: 'show'}));`
In this case I actually get `2 * a` as expected.

Edit:
There actually is a bug in the current toString method of the OperatorNode class.

In line 389 of OperatorNode.js there is a variable "implicit" being set to whatever value you provide to the function within the options object:
` var implicit = options && options.implicit ? options.implicit : 'hide';`

Later on in line 430, this value should be used to check if the Operator sign should be removed, but instead this.implicit is used, which is oftentimes false.
`if (this.implicit && this.getIdentifier() === 'OperatorNode:multiply' && implicit === 'hide') {
        return lhs + ' ' + rhs;
      }`

If you change `this.implicit` to `implicit` it works as expected.

**Comment by josdejong** [comment]
Thanks Beriwan, interesting issue. The option `implicit` indeed works one way: to force showing implicit multiplications as explicit multiplications. 

We could consider extending this option to also render explicit multiplications as implicit when passing `{implicit: 'hide'}`. Not sure if that's easy to do, we need to have the information to decide whether it is safe to render as implicit multiplication which may be tricky. We should also add an option value `{implicit: 'keep'}` then (similar to the option `parenthesis`) which renders implicit and explicit multiplications as is.

**Comment by gwhitney** [comment]
As this seems more of a possible idea for mathjs rather than a specific concrete proposal that could be directly implemented, moving to discussions.
